jax/_src/api.py

```diff
@@ -2814,6 +2814,25 @@ def eval_shape(fun, *args, **kwargs):
   (2000, 1000)
   >>> print(out.dtype)
   float32
+
+  In case the function has static arguments, it should be wrapped
+  using :func:`functools.partial` before passing to :func:`eval_shape`:
+
+  >>> import jax
+  >>> from jax import lax
+  >>> from functools import partial
+  >>> import jax.numpy as jnp
+  >>>
+  >>> x = jax.ShapeDtypeStruct((1, 1, 28, 28), jnp.float32)
+  >>> kernel = jax.ShapeDtypeStruct((32, 1, 3, 3), jnp.float32)
+  >>>
+  >>> out = jax.eval_shape(
+  >>>   partial(lax.conv_general_dilated, window_strides=(1, 1), padding="SAME"),
+  >>> x, kernel)
```

Comment:
The formatting here makes it a bit hard to understand. Maybe split into two lines?
```
>>> conv_same = partial(lax.conv_general_dilated, window_strides=(1, 1), padding="SAME")
>>> out = jax.eval_shape(conv_same, x, kernel)
```

Reply:
Indeed, much better readable.